packages/blocks/src/types.ts

```diff
@@ -0,0 +1,1504 @@
+	 */
+	style?: string;
+
+	/**
+	 * Block styles.
+	 *
+	 * @see {@link https://wordpress.org/gutenberg/handbook/extensibility/extending-blocks/#block-style-variations}
+	 */
+	styles?: BlockStyle[];
+
+	/**
+	 * Optional block extended support features.
+	 *
+	 * @see {@link https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/ Block Supports} on developer.wordpress.org
+	 */
+	supports?: BlockSupports;
+
+	/**
+	 * The gettext text domain of the plugin/block.
+	 *
+	 * @see {@link https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#text-domain}
+	 */
+	textdomain?: string;
+
+	/**
+	 * This is the display title for your block, which can be translated
+	 * with our translation functions.
+	 */
+	title: string;
+
+	/**
+	 * Block transformations.
+	 */
+	transforms?:
+		| {
+				/**
+				 * Transforms from another block type to this block type.
+				 */
+				from?: Transform< Attributes >[];
+				/**
+				 * Transforms from this block type to another block type.
+				 */
+				to?: Transform[];
+				/**
+				 * TODO Undocumented
+				 *
+				 * Used in [factory.js](./api/factory.js)
+				 */
+				supportedMobileTransforms?: string[];
+		  }
+		| undefined;
+
+	/**
+	 * Array of the names of context values to inherit from an ancestor
+	 * provider.
+	 *
+	 * @see {@link https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#context}
+	 */
+	usesContext?: string[];
+
+	/**
+	 * The current version number of the block, such as 1.0 or 1.0.3.
+	 *
+	 * @see {@link https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#version}
+	 */
+	version?: string;
+
+	/**
+	 * Sets attributes on the topmost parent element of the current block.
+	 */
+	getEditWrapperProps?(
+		attrs: Attributes
+	): Record< string, string | number | boolean >;
+
+	/**
+	 * TODO Undocumented, but used in core.
+	 *
+	 * @see `@wordpress/block-library/src/paragraph`
+	 */
+	merge?(
+		attributes: Attributes,
+		attributesToMerge: Attributes
+	): Partial< Attributes >;
+
+	/**
+	 * TODO Undocumented
+	 *
+	 * @internal
+	 */
+	__experimentalLabel(
+		attributes: BlockAttributes,
+		settings: { context: 'accessibility' | 'visual' } // TODO find all context types
+	): string;
+}
+
+/**
+ * The raw structure of a block includes its attributes, inner
+ * blocks, and inner HTML. It is important to distinguish inner blocks from
+ * the HTML content of the block as only the latter is relevant for block
+ * validation and edit operations.
+ *
+ * @public
+ */
+export type BlockRaw = {
+	/**
+	 * Block name
+	 */
+	blockName?: string;
+
+	/**
+	 * Block raw or comment attributes.
+	 */
+	attrs?: Record< string, unknown >;
+
+	/**
+	 * HTML content of the block.
+	 */
+	innerHTML: string;
+
+	/**
+	 * Content without inner blocks.
+	 */
+	innerContent: Array< string | null >;
+
+	/**
+	 * Inner Blocks.
+	 */
+	innerBlocks: BlockRaw[];
+};
```

Comment:
I'd be happy to expand on these descriptions if you want

Reply:
I would greatly appreciate the help

Reply:
maybe I'll try and post some here in a comment which you can absorb into your branch. if I don't do this by Wednesday please feel free to poke me and remind me to do so 😄

Reply:
@dmsnell (👈🏻 poke) haha that reminded of poke in old Facebook 😁

Reply:
see, I knew it would be good to have you remind me, as I got caught up and didn't get to it… doing this today